src/main/java/cpw/mods/niofs/union/UnionPath.java

```diff
@@ -98,28 +112,25 @@ public boolean isAbsolute() {
 
     @Override
     public Path getRoot() {
-        // Found nothing in the docs that say a non-absolute path can't have a root
-        // although this is uncommon. However, other stuff relies on it so leave it
```

Comment:
What is the other stuff that relies on it?

Reply:
That comment got added in the first PR to fix issues with UnionPath. The basic issue was, that at the time, a lot of projects treated relative paths as if they were absolute, relativizing them with other absolute paths and calling getRoot() on it. iirc, when I was playing around with it, I had to make some changes to modlauncher to get forge starting and then it crashed somewhere in FMLLoader. However that was almost 3 years ago and I don't know, how up-to-date that comment still is.

I would assume, it shouldn't really be much af a problem by now because `Jar` now uses regular `JarFileSystem`s when there's only one path (that wasn't the case back then) and code seems to work with them as well (which it didn't when that comment was written).